src/server/editorServices.ts

```diff
@@ -1478,11 +1668,47 @@ namespace ts.server {
                 }
                 if (args.extraFileExtensions) {
                     this.hostConfiguration.extraFileExtensions = args.extraFileExtensions;
+                    // We need to update the project structures again as it is possible that existing
+                    // project structure could have more or less files depending on extensions permitted
+                    this.reloadProjects();
                     this.logger.info("Host file extension mappings updated");
                 }
             }
         }
 
+        /* @internal */
+        closeFileWatcher(watchType: WatchType, project: Project, file: string, watcher: FileWatcher, reason: WatcherCloseReason) {
+            this.logger.info(`FileWatcher:: Close: ${file} Project: ${project ? project.getProjectName() : ""} WatchType: ${watchType} Reason: ${reason}`);
+            watcher.close();
+        }
+
+        /* @internal */
+        addFileWatcher(watchType: WatchType, project: Project, file: string, cb: FileWatcherCallback) {
+            this.logger.info(`FileWatcher:: Added: ${file} Project: ${project ? project.getProjectName() : ""} WatchType: ${watchType}`);
+            return this.host.watchFile(file, (fileName, eventKind) => {
+                this.logger.info(`FileWatcher:: File ${FileWatcherEventKind[eventKind]}: ${file} Project: ${project ? project.getProjectName() : ""} WatchType: ${watchType}`);
+                cb(fileName, eventKind);
+            });
+        }
+
+        /* @internal */
+        closeDirectoryWatcher(watchType: WatchType, project: Project, directory: string, watcher: FileWatcher, flags: WatchDirectoryFlags, reason: WatcherCloseReason) {

```

Comment:
All the parameters other than watcher are for logging purposes. close on the watcher is doesnt take any parameter.